Feat/snowflake tutorial (#1071)
fishtown-analytics/docs.getdbt.com: website/docs/tutorial/1b-setting-up-snowflake.md

id: setting-up-snowflake
---

## Get started with Snowflake

1. To get started, sign up for a free trial by following [this link](https://signup.snowflake.com/). 

2. Then fill out the signup form.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_free_trial_signup_form.png" title="Snowflake Free Trial Signup Form" />
</p>

> 3. Select the Enterprise edition, choose a cloud provider and region, and agree to the terms of service. There are some organizational questions to think about when choosing a cloud provider for a full implementation, which you can read more about [here](https://docs.snowflake.com/en/user-guide/intro-cloud-platforms.html). For the purposes of this setup, all cloud providers and regions will work so choose whichever you’d like. Finally, click "GET STARTED".

Reviewer: **praise:** Love the differentiation here to say, when you are going to fully implement this look here, but pick any for the purposes of this tutorial.

<p align="center">
<Lightbox  src="/img/snowflake_tutorial/snowflake_account_configuration.png" title="Snowflake Account Configuration" />
</p>

4. After submitting the signup form you should receive an email asking you to activate your account. Click on the link in the email and a new tab will open up where you’ll create your username and password. Enter in all that info and click "Get started".

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_account_signup.png" title="Snowflake Account Signup" />
</p>

5. Congrats! Your workspace is ready for some data. Feel free to check out any of the getting started tooltips that Snowflake provides in the UI to familiarize yourself before moving on to the next section.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_workspace.png" title="Snowflake Workspace" />
</p>

## Load sample data into Snowflake

> 1. Now we’re ready for some sample data. The data used here is stored as CSV files in a public S3 bucket and the following steps will guide you through how to prepare your Snowflake account for that data and upload it.

Reviewer: ```suggestion
Now we’re ready for some sample data. The data used here is stored as CSV files in a public S3 bucket and the following steps will guide you through how to prepare your Snowflake account for that data and upload it.
```
suggestion: since this is setting context rather than an action taken by the learner, I'd suggest making this a normal paragraph item.

Reviewer: note: this will have cascading changes for the rest of the list

Author: Sounds good, easy fix

> 2. Run the following commands to create a new virtual warehouse, two new databases (one for raw data, the other for future dbt development), and a `jaffle_shop` schema for your raw data. Feel free to copy/paste from below:

Reviewer 2: Could we link to the how we set up Snowflake doc for more context on why we created these?

Reviewer: +1 that's a great idea.  Here is the link: https://blog.getdbt.com/how-we-configure-snowflake/

Author: Added a sentence mentioning and linking the blog

```sql
create warehouse transforming;

create database raw;

create database analytics;

> create schema raw.jaffle_shop;

Reviewer: ```suggestion
create schema raw.jaffle_shop;

create schema raw.stripe;
```
**suggestion:** moving all the create schema commands here to consolidate warehouse, database, and schema setups

Author: Good call, updated.

```

<p align="center">

> <Lightbox src="/img/snowflake_tutorial/snowflake_create_warehouse_database_and_schema_commands.png" title="Snowflake - Create Warehouse, Database, and Schema Commands" />

Reviewer: idea/question: It looks like the new defaults query editor for snowflake is using this new interface. Do we know if this is the new standard interface for Snowflake?  Would it be worth adjusting the tutorial while we are here with new screenshots and slightly different directions?
<img width="1435" alt="image" src="https://user-images.githubusercontent.com/10554338/152813977-86236df0-4916-46a9-8c6a-671fbd0e3900.png">

Reviewer: new screenshot for this one if you want to use it 
<img width="372" alt="image" src="https://user-images.githubusercontent.com/10554338/152814259-e0f58078-6d5b-4c08-a77d-0e0474db6272.png">

Reviewer 2: @coapacetic We might want to change over to the new UI because it's the standard UI you land in with new accounts. I just recently have to update my quickstart fort his reason

Reviewer: ^^ that's what I was thinking - @bobbybirstockdbtlabs let us know if you need help grabbing some new screenshots / steps.  We likely want to tell the learner to make a new worksheet, etc.
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/10554338/152832823-895ac84b-daaa-42c2-8e6a-8996b5be62ab.png">

Author: Yup, I can definitely switch this up and take new screenshots. It's interesting because for whatever reason, my trials still default to the classic view and make me login again to be able to see everything in "Snowsight". But I'd imagine this will change soon for everyone, better to have them updated now.

	field_delimiter = ','
	skip_header = 1
	)
;
```

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_create_customers_table.png" title="Snowflake - Create Customers Table" />
</p>

4. Create the `jaffle_shop.orders` table and then copy the orders data from S3 into the orders table by following these commands:

```sql
create table raw.jaffle_shop.orders
(
  id integer,
  user_id integer,
  order_date date,
  status varchar,
  _etl_loaded_at timestamp default current_timestamp
);

copy into raw.jaffle_shop.orders (id, user_id, order_date, status)
from 's3://dbt-tutorial-public/jaffle_shop_orders.csv'
file_format = (
	type = 'CSV'
	field_delimiter = ','
	skip_header = 1
	)
;
```

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_create_orders_table.png" title="Snowflake - Create Orders Table" />
</p>

5. Create the stripe schema, the payment table, and copy the payment data into the payment table by following these commands:

```sql
create schema raw.stripe;

create table raw.stripe.payment 
( id integer,
  orderid integer,
  paymentmethod varchar,
  status varchar,
  amount integer,
  created date,
  _batched_at timestamp default current_timestamp

> );

Reviewer: ```suggestion
```sql
create table raw.stripe.payment 
( id integer,
  orderid integer,
  paymentmethod varchar,
  status varchar,
  amount integer,
  created date,
  _batched_at timestamp default current_timestamp
);
```
**suggestion:** move the schema creation to earlier in the tutorial.

Author: Updated

copy into raw.stripe.payment (id, orderid, paymentmethod, status, amount, created)
from 's3://dbt-tutorial-public/stripe_payments.csv'
file_format = (
	type = 'CSV'
	field_delimiter = ','
	skip_header = 1
	)
;
```

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_create_payments_table.png" title="Snowflake - Create Payments Table" />
</p>

6. Great! Your data is loaded and ready to go. Just to make sure, run the following commands to query your data and confirm that you see an output for each one. 

```sql
select * from raw.jaffle_shop.customers;

select * from raw.jaffle_shop.orders;

select * from raw.stripe.payment;
```

Now we’re ready to set up dbt Cloud!

> 

Reviewer: ```suggestion
## Setting up a dbt Cloud project to connection dbt Cloud to Snowflake

There are two ways to connect dbt Cloud and Snowflake. The first option is Partner Connect which provides a streamlined setup between the two.  The second option is to manually connect the two through password authentication.  If you are looking to get started quickly, we recommend **option 1**.  If you are looking to customize your setup from the very beginning, we recommend **option 2**.
```
suggestion: adding in clarification for the two options for connecting dbt Cloud and Snowflake.  This will give context to the learner and allow them to jump to the option that makes the most sense.

Author: Yup yup, added that in and made some slight tweaks.

> ## Set up a project to connect dbt Cloud to Snowflake with Partner Connect

Reviewer: ```suggestion
### Option 1: Connect dbt Cloud and Snowflake with partner connect
```

Author: yup, updated

1. With your Snowflake account up and running with data, we’re ready to connect it with dbt Cloud. We’re going to use [Snowflake Partner Connect](https://docs.snowflake.com/en/user-guide/ecosystem-partner-connect.html) to set up your dbt Cloud account and project. Using Partner Connect will allow you to create a complete dbt account with your [Snowflake connection](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database#connecting-to-snowflake), [a managed repository](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/cloud-using-a-managed-repository), [environments](https://docs.getdbt.com/docs/guides/managing-environments), and credentials. 

> 2. There’s a couple of ways you can access the Partner Connect page depending on if you’re navigating in the classic Snowflake UI or the new UI. 

Reviewer: **praise:** excellent call out for these two options for the learner. Check out my comment in the first section about doing this earlier as well.  They may deprecate the classic version soon which will make this a lot easier.

>   Snowflake Classic UI: If you’re using the classic version of the Snowflake UI, you can click the Partner Connect button in the top bar of your account. From there, click on the dbt tile to open up the connect box. 

Reviewer: ```suggestion
a. Snowflake Classic UI: If you’re using the classic version of the Snowflake UI, you can click the Partner Connect button in the top bar of your account. From there, click on the dbt tile to open up the connect box. 
```
suggestion: nesting this as a substep to get the indentation right.

Author: yea, I was messing with this a bunch before sending it to you for review. Let's give it a shot.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_classic_ui_partner_connect.png" title="Snowflake Classic UI - Partner Connect" />
</p>

> Snowflake New UI: If you’re using the new web interface, you’ll want to click on your name in the upper left hand corner and then click on Partner Connect in the drop down menu.  You can scroll down to find the dbt tile, or search for dbt in the search bar and it will float to the top. Click on the tile to open up the connect box.

Reviewer: ```suggestion
b. Snowflake New UI: If you’re using the new web interface, you’ll want to click on your name in the upper left hand corner and then click on Partner Connect in the drop down menu.  You can scroll down to find the dbt tile, or search for dbt in the search bar and it will float to the top. Click on the tile to open up the connect box.
```
suggestion: similar to above.

Author: 👍

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_new_ui_partner_connect.png" title="Snowflake New UI - Partner Connect" />
</p>

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_partner_connect_box.png" title="Snowflake Partner Connect Box" />
</p>

> 3. Once you’ve clicked on the tile, a connection box will appear that will look slightly different depending on the route you took above, but will contain the same Optional Grant box towards the bottom. In both cases, you’ll want to type in or select the `RAW` and `ANALYTICS` databases. This will grant access for your new dbt user role to each database. 

Reviewer 2: Did we already create those databases earlier?

> <p align="center">

Reviewer 2: Kyle already said this but the screenshots for both UI's. SO Great

<Lightbox src="/img/snowflake_tutorial/snowflake_classic_ui_connection_box.png" title="Snowflake Classic UI - Connection Box" />
</p>

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_new_ui_connection_box.png" title="Snowflake New UI - Connection Box" />
</p>

4. After you’ve entered the database names using either option above, click "Connect". You should see a pop up window similar to the one of the options below. Click Activate.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_classic_ui_activation_window.png" title="Snowflake Classic UI - Actviation Window" />
</p>

<p align="center">
<Lightbox src="/img/snowflake_tutorial/snowflake_new_ui_activation_window.png" title="Snowflake New UI - Activation Window" />
</p>

5. A new tab will be created that will take you to the dbt Cloud website. Here you’ll be asked to create an account name with password, as well as agree to the Terms of Service. Once that’s done, click Complete Registration.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_account_info.png" title="dbt Cloud - Account Info" />
</p>

> 

Reviewer: ```suggestion
6. We have one slight tweak to make to dbt Cloud interface to account for the `analytics` database and `transforming` warehouse created earlier.  Click the hamburger menu in the top left and choose account settings.  Select the project titled, "Partner Connection Trial" and select `snowflake` in the overview table.  Select edit and update the fields `database` and `warehouse` to be `analytics` and `transforming` respectively.
```
This is a slight tweak to make so that partner connect lands the user in the same place as the manual set up.

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/10554338/152818019-1fffc519-de31-451a-ad9a-a90689fcfcb3.png">

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/10554338/152818376-1ee6f8d5-11c6-4fb0-8d5d-d9e0d652d854.png">

Author: Good catch, updated the copy and added the two screenshots.

6. Great! Your dbt Cloud account is now completely setup and connected to your Snowflake trial account with a [managed repository](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/cloud-using-a-managed-repository). You can skip to the "How to start developing in dbt Cloud" section to get started in the IDE. 

> ## Set up a project to connect dbt Cloud to Snowflake Without Partner Connect

Reviewer: ```suggestion
### Option 3: Connect dbt Cloud to Snowflake manually
```

Author: fixed

> 1. With your Snowflake account up and running with data, we’re ready to connect it with dbt Cloud. We’ll start by going to [https://cloud.getdbt.com/signup/](https://cloud.getdbt.com/signup/) to start the signup process. Fill out all the information and click "Create my account".

Reviewer 2: ```suggestion
1. With your Snowflake account up and running with data, we’re ready to connect it with dbt Cloud. We’ll start by going to [dbt Cloud Signup Form](https://cloud.getdbt.com/signup/) to start the signup process. Fill out all the information and click "Create my account".
```

Author: yup

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_account_registration.png" title="dbt Cloud - Account Registration" />
</p>

2. Once your account is created you’ll get a verification email. When you click the "Verify Email Address" button in the email you should be redirected to your new dbt Cloud account.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_setup.png" title="dbt Cloud - Setup" />
</p>

3. Clicking Continue will send you to start setting up the database connection. Click Snowflake.

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_connect_snowflake.png" title="dbt Cloud - Connect Snowflake" />
</p>

> 4. To create the connection we’ll start by inputting the account information. Your account is going to be the url of your Snowflake trial account up to the period before snowflakecomputing.com. So, if this is the full url of my trial account: `oq65696.west-us-2.azure.snowflakecomputing.com`

Reviewer 2: Could you link to this doc? It's surprisingly confusing https://docs.snowflake.com/en/user-guide/admin-account-identifier.html

Author: Yup yup

10. Now it’s time to create your Git repository. For this setup we’re going to stick with a dbt Cloud managed repository, but feel free to create your own repo and connect it to your git provider of choice if you’d like to manage it yourself. To create the managed repository, make sure that `Managed` is selected from the list of repos/options and then type in the name that you’d like to give the repo. Once complete, click "Create".

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_set_up_a_repository.png" title="dbt Cloud - Set Up a Repository" />
</p>

It should take a few seconds to create the repo and once it’s complete you’ll see a Successfully imported repository message and you’ll be able to click Continue. 

11. You can skip the final step of inviting other users to your account and click Skip & Complete to finish the setup. You’ll be able to go back and add users at any point within your account. 

12. Congrats! Your account is officially set up and ready to rock. 

## Start developing in dbt Cloud

> 1. On the welcome page click "Start Developing", which will take you to Cloud IDE.

Reviewer: quick-fix: Let's update this to use the hamburger menu on the left.  When I went through the partner connect option, I didn't see the green 'start developing' button.
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/10554338/152816638-4fb53978-ef17-453b-8b8f-aa5e0af96fcf.png">

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_welcome_page.png" title="dbt Cloud - Welcome Page" />
</p>

2. After the Cloud IDE loads, click the "Initialize your project" button in the upper left hand corner of the screen to initialize and create the directory structure within your project.

> <p align="center">

Reviewer: praise: awesome move with the blue box around the button to click.  This little things goes a long way for making this clear to the learner.

<Lightbox src="/img/snowflake_tutorial/dbt_cloud_initialize_your_project.png" title="dbt Cloud - Initialize Your Project" />
</p>

> 3. Next you’ll want to commit all the newly created folders and files to your main branch. Click "commit", enter a brief message, and then click "Commit" again in the pop up window.

Reviewer: ```suggestion
3. Next you’ll want to commit all the newly created folders and files to your main branch. Click "commit", enter a brief message, and then click "commit" again in the pop up window.
```

Author: Fixed

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_click_commit.png" title="dbt Cloud - Click Commit" />
</p>

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_write_commit_message.png" title="dbt Cloud - Write Commit Message" />
</p>

4. Next we’re going to make sure that you can query the data we loaded earlier from within the IDE. Enter the following code into a tab in the IDE and click preview to view the results:
`select * from raw.jaffle_shop.customers`

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_preview_results.png" title="dbt Cloud - Preview Results" />
</p>

5. Alright, we’re ready to run our first dbt command. In the run bar at the bottom of the screen enter `dbt run` and click "Enter".

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_dbt_run.png" title="dbt Cloud - dbt run" />
</p>

> When you expand the run bar you should see something similar to the below. This run should have created the two example models in your development schema in Snowflake. By clicking on each model you should see more details about each run in the drop down menu. Next, go into your trial Snowflake account and open up the `ANALYTICS` database. You should be able to see that your development schema was created by this run and within that schema both of the models that you just ran.

Reviewer: important-note: If using partner connect, this builds into `pc_dbt_db` by default rather than analytics.  Let's add a very quick change from `pc_dbt_db` to `analytics` in the partner connect set up above.

Author: 👍 definitely

<p align="center">
<Lightbox src="/img/snowflake_tutorial/dbt_cloud_run_results.png" title="dbt Cloud - Run Results" />
</p>

Awesome!  You are all set to leverage the power of dbt Cloud and Snowflake together.  As a recap, we just completed the following:

> 
> * Set up a Snowflake trial account
> 
> * Loaded training data into your Snowflake account
> 
> * Created a dbt Cloud account, either through Partner Connect or through the account flow
> 
> * Connected dbt Cloud and Snowflake
> 
> * Set up the dbt Cloud IDE, queried data, and did your first dbt run

Reviewer: ```suggestion
* Set up a Snowflake trial account
* Loaded training data into your Snowflake account
* Created a dbt Cloud account, either through Partner Connect or through the account flow
* Connected dbt Cloud and Snowflake
* Set up the dbt Cloud IDE, queried data, and did your first dbt run
```
suggestion: making the list a bit more concise for readability.

Author: 👍
